wgpu-hal/src/vulkan/device.rs

```diff
@@ -1414,7 +1414,11 @@ impl crate::Device for super::Device {
         //
         // https://github.com/gfx-rs/wgpu/issues/6867
         if let Some(label) = desc.label {
+            let l = self.shared.sampler_cache.lock();
```

Comment:
nitpick: I think we should use the first `lock` call's guard from immediately above, and extend its scope, so we don't have two lock regions for the same lock in series. Otherwise, I think this commit can mostly be the same?

Will push a `fixup!` for this shortly.

Reply:
Pushed [65c9f0e](https://github.com/gfx-rs/wgpu/pull/8043/commits/65c9f0e51be18acda21d9526e350b7ad468a0067).